<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1 plus MathML 2.0//EN" "http://www.w3.org/Math/DTD/mathml2/xhtml-math11-f.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><!--This file was converted to xhtml by LibreOffice - see https://cgit.freedesktop.org/libreoffice/core/tree/filter/source/xslt for the code.--><head profile="http://dublincore.org/documents/dcmi-terms/"><meta http-equiv="Content-Type" content="application/xhtml+xml; charset=utf-8"/><title xml:lang="en-US">- no title specified</title><meta name="DCTERMS.title" content="" xml:lang="en-US"/><meta name="DCTERMS.language" content="en-US" scheme="DCTERMS.RFC4646"/><meta name="DCTERMS.source" content="http://xml.openoffice.org/odf2xhtml"/><meta name="DCTERMS.issued" content="2020-11-07T17:18:50.759121239" scheme="DCTERMS.W3CDTF"/><meta name="DCTERMS.modified" content="2021-05-06T17:23:23.408134043" scheme="DCTERMS.W3CDTF"/><meta name="DCTERMS.provenance" content="" xml:lang="en-US"/><meta name="DCTERMS.subject" content="," xml:lang="en-US"/><link rel="schema.DC" href="http://purl.org/dc/elements/1.1/" hreflang="en"/><link rel="schema.DCTERMS" href="http://purl.org/dc/terms/" hreflang="en"/><link rel="schema.DCTYPE" href="http://purl.org/dc/dcmitype/" hreflang="en"/><link rel="schema.DCAM" href="http://purl.org/dc/dcam/" hreflang="en"/><style type="text/css">
    @page {  }
    table { border-collapse:collapse; border-spacing:0; empty-cells:show }
    td, th { vertical-align:top; font-size:12pt;}
    h1, h2, h3, h4, h5, h6 { clear:both;}
    ol, ul { margin:0; padding:0;}
    li { list-style: none; margin:0; padding:0;}
    /* "li span.odfLiEnd" - IE 7 issue*/
    li span. { clear: both; line-height:0; width:0; height:0; margin:0; padding:0; }
    span.footnodeNumber { padding-right:1em; }
    span.annotation_style_by_filter { font-size:95%; font-family:Arial; background-color:#fff000;  margin:0; border:0; padding:0;  }
    span.heading_numbering { margin-right: 0.8rem; }* { margin:0;}
    .P1 { font-size:14pt; font-family:DejaVu Sans; writing-mode:horizontal-tb; direction:ltr; }
    .P10 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; font-style:normal; text-shadow:none; text-decoration:none ! important; font-weight:normal; }
    .P11 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P12 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P13 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; }
    .P14 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P15 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; font-weight:bold; }
    .P16 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; }
    .P17 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; font-style:italic; }
    .P18 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; font-style:italic; }
    .P19 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; font-style:italic; }
    .P20 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; font-style:italic; }
    .P21 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; }
    .P22 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; }
    .P23 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; text-align:left ! important; }
    .P24 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P25 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P26 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P27 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P3 { font-size:12pt; font-family:DejaVu Sans; writing-mode:horizontal-tb; direction:ltr; }
    .P4 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P5 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P6 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; font-style:italic; }
    .P7 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P8 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; }
    .P9 { font-size:12pt; font-family:Lato Medium; writing-mode:horizontal-tb; direction:ltr; font-style:normal; text-shadow:none; text-decoration:none ! important; font-weight:normal; }
    .Table1 { width:25.4cm; margin-left:-0.106cm; margin-top:0cm; margin-bottom:0cm; margin-right:auto;}
    .Table2 { width:25.7cm; float:none; }
    .Table1_A1 { vertical-align:bottom; padding-left:0.049cm; padding-right:0.049cm; padding-top:0cm; padding-bottom:0cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-style:none; border-top-width:thin; border-top-style:solid; border-top-color:#000000; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table1_A2 { vertical-align:bottom; padding-left:0.049cm; padding-right:0.049cm; padding-top:0cm; padding-bottom:0cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-style:none; border-top-style:none; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table1_C1 { vertical-align:bottom; padding-left:0.049cm; padding-right:0.049cm; padding-top:0cm; padding-bottom:0cm; border-width:thin; border-style:solid; border-color:#000000; }
    .Table1_C2 { vertical-align:bottom; padding-left:0.049cm; padding-right:0.049cm; padding-top:0cm; padding-bottom:0cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-width:thin; border-right-style:solid; border-right-color:#000000; border-top-style:none; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table2_A1 { padding:0.097cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-style:none; border-top-width:thin; border-top-style:solid; border-top-color:#000000; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table2_A2 { padding:0.097cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-style:none; border-top-style:none; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table2_H1 { padding:0.097cm; border-width:thin; border-style:solid; border-color:#000000; }
    .Table2_H2 { padding:0.097cm; border-left-width:thin; border-left-style:solid; border-left-color:#000000; border-right-width:thin; border-right-style:solid; border-right-color:#000000; border-top-style:none; border-bottom-width:thin; border-bottom-style:solid; border-bottom-color:#000000; }
    .Table1_A { width:3.946cm; }
    .Table1_B { width:5.341cm; }
    .Table1_C { width:16.113cm; }
    .Table2_A { width:3.212cm; }
    .Table2_H { width:3.216cm; }
    .Internet_20_link { color:#000080; text-decoration:underline; }
    .T10 { font-family:Lato Medium; }
    .T7 { color:#ff0000; text-decoration:underline; }
    /* ODF styles with no properties representable as CSS */
    .Table1.1 .Table1.5 .Table1.6 .Table1.9 .T1 .T5 .T6 .T8  { }
    </style></head><body dir="ltr" style="max-width:29.7cm;margin-top:2cm; margin-bottom:2cm; margin-left:2cm; margin-right:2cm; "><p class="P12">The table below dissects the first entry in our VCF file, which looks like this(added some spaces for clarity):</p><p class="P12"> </p><p class="P12">#CHROM<span> POS</span><span> ID</span><span> REF</span><span> ALT</span><span> QUAL</span><span> FILTER</span><span> INFO</span><span> FORMAT</span><span> NA12878</span><span> NA12891</span><span> NA12892</span></p><p class="P12">1<span> 10403</span><span> .</span><span> ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC</span><span> A</span><span> 120.15</span><span> PASS</span><span> AC=2;AF=0.333;AN=6;BaseQRankSum=0.753;ClippingRankSum=4.579;DP=134;FS=0.000;MLEAC=2:</span></p><p class="P12">MLEAF=0.333;MQ=39.50;MQ0=0;MQRankSum=0.851;PG=0,0,0;QD=1.47;ReadPosRankSum=-0.531;</p><p class="P12">VQSLOD=3.38;culprit=FS<span> GT:AD:DP:GQ:PL:PP:TP</span><span> </span><span> -0/1:34,5:39:75:75,0,1282:75,0,1306:24</span><span> 0/0:37,3:40:24:0,24,1615:0,24,1615:24</span><span> 0/1:15,4:19:99:86,0,465:110,0,465:24</span></p><p class="P12"> </p><p class="P12">The Variable “INFO” contains multiple fields <span class="T8">(AKA Extensible meta-data)</span>, <span class="T5">each </span>separated <span class="T5">by</span> a semi-colon.</p><p class="P14"><span class="T5">The Variable “FORMAT” </span><span class="T8">again </span><span class="T1">contains multiple fields, </span><span class="T5">each </span><span class="T1">separated </span><span class="T5">by </span><span class="T1">a semi-colon </span><span class="T5">and it directly refers to the format of the subsequent fields: “NA12878”, “NA12891”, “NA12892”. Each of these is a different sample with the values of the variables described by FORMAT</span></p><p class="P14"> </p><table border="0" cellspacing="0" cellpadding="0" class="Table1"><colgroup><col width="172"/><col width="233"/><col width="704"/></colgroup><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A1"><p class="P15">Variable/Col Heading</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A1"><p class="P15">Example</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C1"><p class="P15">Meaning</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">CHROM</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">1</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">Chromosome</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">POS</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">10403</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">position on the chromosome</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">ID</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">.</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">RSID from “www.<span class="T1">s</span>npedia.com”</p></td></tr><tr class="Table15"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">REF</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">Refer<span class="T1">e</span>nce sequence</p></td></tr><tr class="Table16"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">ALT</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">A</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">Alternative sequence</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">QUAL</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">120.15</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16"> </p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">FILTER</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">PASS</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16"> </p></td></tr><tr class="Table19"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P21">INFO</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P19">AC=2</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P21">allele count in genotypes, for each ALT allele, in the same order as listed </p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">AF=0.333</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">allele frequency for each ALT allele in the same order as listed (use this when estimated from primary data, not called genotypes) </p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">AN=6</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">  total number of alleles in called genotypes </p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">BaseQRankSum=0.753</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Z-score from Wilcoxon rank sum test of Alt Vs. Ref base qualities</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">ClippingRankSum=-4.579</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Z-score From Wilcoxon rank sum test of Alt vs. Ref number of hard clipped bases</p></td></tr><tr class="Table19"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">DP=134</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Approximate read depth; some reads may have been filtered</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">FS=0.000</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Phred-scaled p-value using Fisher's exact test to detect strand bias</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">MLEAC=2</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), f or each ALT allele, in the same order as listed</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">MLEAF=0.333</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF),  for each ALT allele, in the same order as listed"</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">MQ=39.50</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">RMS mapping quality </p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">MQ0=0</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Total Mapping Quality Zero Reads</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">MQRankSum=0.851</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">PG=0,0,0</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Genotype Likelihood Prior</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">QD=1.47</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Variant Confidence/Quality by Depth</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">ReadPosRankSum=-0.531</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">score from Wilcoxon rank sum test of Alt vs. Ref read position bias</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P22"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P20">VQSLOD=3.38</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P22">Log odds ratio of being a true variant versus being false under the trained gaussian mixture model</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P21"> </p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P19">culprit=FS</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P21">The annotation which was the worst performing in the Gaussian mixture model, likely the reason wh y the variant was filtered out</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P16">FORMAT</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17">GT: AD: DP: GQ: PL: PP: TP</p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P16">These refer to the format of the samples:</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P13">NA12878</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P18"> </p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P23">These three “NA….” are samples and contain meta-data in the format</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P13">NA12891</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17"> </p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P23">specified in FORMAT</p></td></tr><tr class="Table11"><td style="text-align:left;width:3.946cm; " class="Table1_A2"><p class="P13">NA12892</p></td><td style="text-align:left;width:5.341cm; " class="Table1_A2"><p class="P17"> </p></td><td style="text-align:left;width:16.113cm; " class="Table1_C2"><p class="P23">See the table below for a dissection of this.</p></td></tr></table><p class="P11"> </p><p class="P11"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25"> </p><p class="P25">FORMAT field applied to the samples.</p><p class="P25"> </p><table border="0" cellspacing="0" cellpadding="0" class="Table2"><colgroup><col width="140"/><col width="140"/><col width="140"/><col width="140"/><col width="140"/><col width="140"/><col width="140"/><col width="141"/></colgroup><tr class="TableLine94633661316192"><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P5">FORMAT</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">GT</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">AD</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">DP</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">GQ</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">PL</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A1"><p class="P6">PP</p></td><td style="text-align:left;width:3.216cm; " class="Table2_H1"><p class="P6">TP</p></td></tr><tr class="TableLine94633661339392"><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Meaning</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P7">Genotype</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P7">Allellic depths  for the ref and alt alleles in the order listed</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Approximate read depth (reads with MQ=255 or with bad mates are filtered)</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Genotype Quality</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Phred-scaled Posterior Genotype Probabilities</p></td><td style="text-align:left;width:3.216cm; " class="Table2_H2"><p class="P5">Phred score of the genotype combination and phase given that the genotypes are correct</p></td></tr><tr class="TableLine94633661341408"><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">Sample:  <span class="T5">NA12891</span></p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">0/0</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">37.5</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">40</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P5">24</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P9">0,24,1615</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P9">0,24,1615</p><p class="P26"> </p></td><td style="text-align:left;width:3.216cm; " class="Table2_H2"><p class="P5">24</p></td></tr><tr class="TableLine94633661342928"><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">Sample:</p><p class="P10">NA12878</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P4">-<span class="T6">0/1</span></p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">34.5</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">39</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">75</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">75,0,1282</p></td><td style="text-align:left;width:3.212cm; " class="Table2_A2"><p class="P8">75,0,1306</p></td><td style="text-align:left;width:3.216cm; " class="Table2_H2"><p class="P8">24</p></td></tr></table><p class="P11"> </p><p class="P27">Other format fields can appear, for example in the second observation/ row in our file:</p><p class="P27">1<span> 10433</span><span> .  A ACCCTAACCCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTACCCTAACCCTAACCCTACCCTAACCCTAACCCTAACCCTAACCCCTAAC</span><span> 207.12</span><span> VQSRTrancheINDEL97.00to99.00 AC=2;AF=0.333;AN=6;BaseQRankSum=0.666;ClippingRankSum=1.160;DP=94;FS=5.880;MLEAC=2;MLEAF=0.333;MQ=38.44;MQ0=0;MQRankSum=0.419; PG=0,0,0; QD=2.76;ReadPosRankSum=0.026;VQSLOD=0.094;culprit=QD GT:AD:DP:<span class="T7">FT:</span>GQ:PL:PP:TP</span><span> 0/1:7,1:8:LowConf:1:1,0,661:1,0,700:3</span><span> 0/1:5,9:14:PASS:99:244</span></p><p class="P27">,0,481:248,0,484:3<span> 0/0:4,1:5:PASS:39:0,39,494:0,39,497:3</span></p><p class="P24"> </p><p class="P24">FT = Genotype-level filter</p><p class="P3"><span class="T10">Further descriptions at: </span><a href="https://en.wikipedia.org/wiki/Variant_Call_Format" class="Internet_20_link"><span class="T10">https://en.wikipedia.org/wiki/Variant_Call_Format</span></a></p><p class="P24">Nice graphical abstract of a VCF file: http://vcftools.sourceforge.net/VCF-poster.pdf</p></body></html>