---
title: "LD_Thinning_pcadapt"
author: "Georgina Robertson"
date: "05/06/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Linkage Disequilibrium (LD) thinning
Linkage Disequilibrium can affect ascertainment of population structure.  
In pcadapt, there is an option to compute PCs after SNP thinning. SNP thinning make use of two parameters: window size (default 200 SNPs) and r2 threshold (default 0.1). The genome scan is then performed by looking at associations between all SNPs and PCs ascertained after SNP thinning.  
```{r}
library(pcadapt)
```

### This is for chromosome 1
```{r}
genotypes1 <- read.table("./ByChrom/chr1_snp_matrix.txt", header = F)
genotypes1 <- as.matrix(genotypes1)
# to get it to work you have to remove the sample and loci label!
genotypes1_1 <- genotypes1[,-(1:2)]
```

```{r}
mat1 <- read.pcadapt(genotypes1_1, type = "pcadapt")
res <- pcadapt(mat1, K = 20)
```

```{r}
plot(res, option = "screeplot")
```

Picked K = 3
```{r}
res <- pcadapt(mat1, K =3)
plot(res)
```

To evaluate if LD might be an issue for your dataset, we recommend to display the loadings (contributions of each SNP to the PC) and to evaluate if the loadings are clustered in a single or several genomic regions.
```{r}
par(mfrow = c(2, 2))
for (i in 1:3)
  plot(res$loadings[, i], pch = 19, cex = .3, ylab = paste0("Loadings PC", i))
```
???

```{r}
res <- pcadapt(mat1, K = 20, LD.clumping = list(size = 200, thr = 0.1))
plot(res, option = "screeplot")
```

```{r}
res <- pcadapt(mat1, K = 3, LD.clumping = list(size = 800, thr = 0.1))
par(mfrow = c(1, 2))
for (i in 1:2)
  plot(res$loadings[, i], pch = 19, cex = .3, ylab = paste0("Loadings PC", i))
```
What is this showing? 

```{r}
plot(res)
```
???