---
title: "pcadapt_WF"
author: "Georgina Robertson"
date: "21/05/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Using pcadapt to detect local adaptation
Following the walkthrough from:
https://bcm-uga.github.io/pcadapt/articles/pcadapt.html
```{r}
# install.packages("pcadapt")
library(pcadapt)
```

## Read data into R:
```{r}
path_to_file <- "FST/drought_pops_pcadapt.bed"
filename <- read.pcadapt(path_to_file, type = "bed")
```

## Choosing number of PCs (K)
Perform PCA:
```{r}
x <- pcadapt(input = filename, ploidy = 1, K =20)
```

Scree plots:
```{r}
plot(x, option = "screeplot")
# saved as scree_plot_k20 in IMGS folder
```
Looks like first two PCs explain most of variance, confirmed below too.
```{r}
plot(x, option = "screeplot", K=10)
# saved as scree_plot_k10 in IMGS folder
```

## Score Plot

Read in the sorted poplist
```{r}
temp <- read.delim("data_sheets/populationIDs_for_droughtpopsVCF.txt", header = F)

poplist.names <- temp$V1
```

## PCA

```{r}
plot(x, option = "scores", pop = poplist.names)
# saved as Dpops_PCA_1_2.png in IMGS folder
```


Make an interactive plot for closer examination
```{r}
library(plotly)
```

```{r}
interative_drought_PCA <- plot(x, option = "scores", pop = poplist.names, plt.pkg = "plotly")
# this interactive version saved as "Pre_Post_Drought_PCA.html" in IMGS folder
```

## Computing the test statistic based on PCA

K = 2 was optimal PCs for our data
```{r}
summary(x)
```
* scores is a (n,K) matrix corresponding to the projections of the individuals onto each PC.
* singular.values is a vector containing the K ordered square root of the proportion of variance explained by each PC.
* loadings is a (L,K) matrix containing the correlations between each genetic marker and each PC.
* zscores is a (L,K) matrix containing the z-scores.
* af is a vector of size L containing allele frequencies of derived alleles where genotypes of 0 are supposed to code for homozygous for the reference allele.
* maf is a vector of size L containing minor allele frequencies.
* chi2.stat is a vector of size L containing the rescaled statistics stat/gif that follow a chi-squared distribution with K degrees of freedom.
* gif is a numerical value corresponding to the genomic inflation factor estimated from stat.
* pvalues is a vector containing L p-values.
* pass A list of SNPs indices that are kept after exclusion based on the minor allele frequency threshold.
* stat is a vector of size L containing squared Mahalanobis distances by default.

All of these elements are accessible using the $ symbol. For example, the p-values are contained in x$pvalues.

## Manhattan Plot
displays −log10 of the p-values.
```{r}
plot(x , option = "manhattan")
```
I'll be honest, this doesn't look very meaningful to me.
note: look into what this means

## Q-Q plot
The expected uniform distribution of the p-values using a Q-Q plot
```{r}
plot(x, option = "qqplot")
```
## Histograms of the test statistic and of the p-values
A histogram of p-values confirms that most of the p-values follow an uniform distribution. The excess of small p-values indicates the presence of outliers.
```{r}
hist(x$pvalues, xlab = "p-values", main = NULL, breaks = 50, col = "blue")
```

The presence of outliers is also visible when plotting a histogram of the test statistic Dj.
```{r}
plot(x, option = "stat.distribution")
```

## Choosing a cutoff for outlier detection
To provide a list of outliers and choose a cutoff for outlier detection, there are several methods that are listed below from the less conservative one to the more conservative one.

q-values
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("qvalue")
```
```{r}
library(qvalue)
```
For a given α (real valued number between 0 and 1), SNPs with q-values less than alpha will be considered as outliers with an expected false discovery rate bounded by α. The false discovery rate is defined as the percentage of false discoveries among the list of candidate SNPs. Here is an example of how to provide a list of candidate SNPs for the geno3pops data, for an expected false discovery rate lower than 10%
```{r}
qval <- qvalue(x$pvalues)$qvalues
alpha <- 0.1
outliers <- which(qval < alpha)
length(outliers)
```

Benjamini-Hochberg Procedure

```{r}
padj <- p.adjust(x$pvalues,method="BH")
alpha <- 0.1
outliers <- which(padj < alpha)
length(outliers)
```

Bonferroni correction
```{r}
padj <- p.adjust(x$pvalues,method="bonferroni")
alpha <- 0.1
outliers <- which(padj < alpha)
length(outliers)
```

## Linkage Disequilibrium (LD) thinning

Need to remake a SNP matrix for this
snp_matrix.rmd

# Work in progress, TBC...