# Z.tritici

## Contents

### Folders:
* IMGS
  - Various plots and images generated by different pipelines

* data_sheets
  - all source metadata
  - generated subsets and tables written to file (final versions)

### Files:
* AUS350.Rmd
  - Pipeline for making the VCF that include drought populations and exclude Hamilton cluster

* Activity_log
  - Week to week goals and actions

* Dissecting VCF files
  - Document i made a while ago detailing the typical contents of a vcf file

* FST_file_prep.Rmd
  - Tidying FST output file, re-writes to file (/data_sheets/fst.file.csv)
  - Merges FST output with loci files (produced from SNP matrix pipeline) to link FST values with genomic loci, and   by-chromosome SNP index (/data_sheets/FST_SNPindex.csv)

* FST_summary.Rmd
  -  Boxplot of summary FST values by chrom, whole genome

* High_Variation_FST.Rmd
  - This file uses the FST output from VCFTools to look at snps on a per chromosone basis which have high variation (FST >= 80, or 90)

* High_n_Low_Plot
  - Playing around with high and low variation snps from the FST data
 
* LD_Calc_VCFtools.Rmd
  - workflow for splitting VCF by population and calculating LD using VCFtools
  - splitting the resulting two population LD files by chromosome
  
* LD_Thinning.Rmd /LD_Thinning.html
  - LD thinning workflow as seen in the pcadapt workflow example

* LD_binning.Rmd
  - Workflow for producing a by pop and by chrom LD decay plot

* Least_Variation.Rmd
  - Similar to "High_Variation.Rmd" but looking at snps with FST <0.05
  - Hoping to find highly conserved domains

* Mapping.rmd
  - Plotting where are our samples are from and thier relative sample size

* Notes.md / Notes.html
  - Notes, reminders, correspondence. Stuff I might need to remember.

* PlotFilteredSNPs.rmd
  - PLots of various quality metrics from the filtered VCF of all isolates
    - From GATK pipeline which combined invidicual VCFs into one large VCF

* REHH.Rmd
  - Work in progress

* Run_filter_may2020.sh
  - GATK pipeline

* amendingfile.R
  - script that modified GATK output VCF
  - used in PlotFilteredSNPs.rmd

* byChrom_snp_matrix.Rmd
  - This document outlines how to create the appropriately formatted matrices for downstream analysis
    - individual steps (deprecated)
    - looped version that automates the above

* drought_status_pop.Rmd
  - pipeline for indicating drought status per isolate and re-ordering the VCF file accordingly

* high_missing_indv.Rmd
  - This removes the 3 individuals that evaded the initial vcf filter (due to being only very slightly above lower threshhold)
  - Then recodes the main VCF

* pcadapt_workflow.Rmd
  - Using package pcadapt to perform genome scans for selection based on principal component analysis 
  - refs: https://bcm-uga.github.io/pcadapt/articles/pcadapt.html , https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12592

* pheno_data
  - playing around with the phenotyped samples

* preparation_for_pcadapt.Rmd
  - assigning population group identifiers to samples as ordered in VCF

* readabe_pcadapt.md
  - as  pcadapt.Rmd, but more readable here.

* readformat_VCF_FILTER_PCA_commented_G.md
  - this is the "Generalised_VCF_filter_PCA" modified and commented by me
  - in a nice readable format that displays well on github

* snp_matrix.rmd
  - workflow for producing and labelling a snp matrix

* snpmatrix_heatmaps.Rmd
  - Using the snp matrices to produce heatmaps

