---
title: "Min_Variation_FST"
author: "Georgina Robertson"
date: "10/06/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Do the regions of no variation co-incide with highly conversrved genes?

Reading in file that contains SNP index and FST data which was made in the pipeline: "SnpIndex_FST.Rmd"
```{r}
FST_SnpIndex <- read.csv("./data_sheets/FST_SNPindex.csv")
FST_SnpIndex <- FST_SnpIndex[,-1]
```

```{r}
lowest <- FST_SnpIndex[FST_SnpIndex$FST <=0.05, ]
```

```{r}
print(unique(lowest$Chrom))
```
All the chroms have these regions.
Will only look at the core chromosomes

```{r}
chr1Lowest <- lowest[lowest$Chrom==1,]
chr2Lowest <- lowest[lowest$Chrom==2,]
chr3Lowest <- lowest[lowest$Chrom==3,]
chr4Lowest <- lowest[lowest$Chrom==4,]
chr5Lowest <- lowest[lowest$Chrom==5,]
chr6Lowest <- lowest[lowest$Chrom==6,]
chr7Lowest <- lowest[lowest$Chrom==7,]
chr8Lowest <- lowest[lowest$Chrom==8,]
chr9Lowest <- lowest[lowest$Chrom==9,]
chr10Lowest <- lowest[lowest$Chrom==10,]
chr11Lowest <- lowest[lowest$Chrom==11,]
chr12Lowest <- lowest[lowest$Chrom==12,]
chr13Lowest <- lowest[lowest$Chrom==13,]
```

```{r}
library(ggplot2)
```

Will look at chr 5, as this has a very interesting heatmap
```{r}
dim(chr5Lowest)
# =4811

c12Low1<- chr12Lowest[1000:1500,]
ggplot()+
geom_col(data = chr5Lowest, aes(x=SNPindex, y = FST), colour = "red") +
  xlab("loci")+
  ylab("fst")
```


```{r}
# determine the loci range of interest
lower <- head(chr13Lowest$POS, n=1)
upper <- tail(chr13Lowest$POS, n=1)

# this puts the info into a vector that can be copy pasted to ensemble
chr13conserved <- paste(chr13Lowest$CHROM[[1]],":", lower, "-", upper, sep = "")

# get the loci from the loci list for the snp matrix
chromosome13 <- read.csv("./Processed_SNP_matrices/chrom_13_loci.csv")

# extract SNP index at the beginning and the end of the region
rangestart <- chromosome13[chromosome13$position==lower, 1]
rangeend <- chromosome13[chromosome13$position==upper, 1]
print(rangestart)
print(rangeend)

# save this for later
chr13conservedSNPindex <- paste(chr13Lowest$CHROM[[1]], rangestart, rangeend, sep = ":")
print(chr13conservedSNPindex)

rm(lower, upper, rangestart, rangeend, chromosome5)
```

I'm quite interested in the 9000-11000 snp region in the heatmap. This is large region of 


```{r}
library(data.table)
setkey()
```
```{r}
```

