---
title: "EXP-EHH"
author: "Georgina Robertson"
date: "15/06/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Cross population extended haplotype homozygosity (XP-EHH)

Further info
> * https://rdrr.io/cran/rehh/man/ies2xpehh.html
* https://www.biostars.org/p/170385/
* https://storage.googleapis.com/plos-corpus-prod/10.1371/journal.pone.0044751/1/pone.0044751.s011.pdf?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=wombat-sa%40plos-prod.iam.gserviceaccount.com%2F20210624%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20210624T103647Z&X-Goog-Expires=86400&X-Goog-SignedHeaders=host&X-Goog-Signature=8c71380986df233206270d523b7794a4278f05dc1a3b8653a8fee4de8ff68b986c74a549941d7acca2d1ba6e01eb44843268f17c6a6084b53df33211a736205cd982f51667e0a4ddc47de72f5950bebc913ee9cf84beff02e80f50765a3308609717c6745b8f576ed2cdb6b80ce6441849fec19534a27fc66cae5d9966a3ffbcc308b5f9ba76b86667df28fb236315466dbac4c5ddd7f8e9abaec7688b7d0dff6d824c8888cb9ac670d0b5e1d4b7c3db03dbcb676921db880db103d5c838f3be3f359ca9a839cba0eb608a1cd9e144d35fc6adb7beca1247d306ceeb900ae2f0c3deef3ac19e2ef35f64f8c770cfd3b46e1706a9512765e8f5413528e8992e35

Required Libraries
```{r}
library(rehh)
library(data.table)
```

### Individual steps

1) Read in the vcf files
Pre and Post populations for the same chromosome
```{r}
chr1PRE <- data2haplohh(hap_file = "./ByPop_Chrom/Chr_01.pre.recode.vcf", vcf_reader = "data.table", verbose = F, polarize_vcf = F)

chr1POST <- data2haplohh(hap_file = "./ByPop_Chrom/Chr_01.post.recode.vcf", vcf_reader = "data.table", verbose = F, polarize_vcf = F)
```

2) Convert haplohh objects to required data frames
```{r}
scan_pop1 <- scan_hh(chr1POST)
scan_pop2 <- scan_hh(chr1PRE)
```

3) Compute XP-EHH 
```{r}
chr1XP_EHH <- ies2xpehh(
  scan_pop1,
  scan_pop2,
  popname1 = "pre",
  popname2 = "post",
  min_nhaplo = NA,
  standardize = TRUE,
  include_freq = FALSE,
  p.side = NA,
  p.adjust.method = "none",
  verbose = TRUE
)
```

4) Write to file (optional)
```{r}
write.csv(chr1XP_EHH, file = "./REHH/chr1XPEHH.csv")
```

### Loop method
Loops through VCF files and perform above steps

1) Store the file paths as variables
```{r}
# read in list of files for each pop
filesPRE <- list.files(path ="./ByPop_Chrom/", pattern = "*.pre.recode.vcf", full.names = T, recursive = T)
filesPOST <- list.files(path ="./ByPop_Chrom/", pattern = "*.post.recode.vcf", full.names = T, recursive = T)

# convert the above lists to dataframes so they can be called in loop
filesPRE <- as.data.frame(filesPRE)
filesPRE$chrom <- c(1:16, 19:21)
filesPOST <- as.data.frame(filesPOST)
filesPOST$chrom <- c(1:16, 19:21)
```

2) Loop through the files, perform the analysis for each pair, write to csv
```{r}
for (i in 1:19) {
  
  prepop  <- data2haplohh(hap_file = filesPRE$filesPRE[i],
          vcf_reader = "data.table", verbose = F, polarize_vcf = F)
  postpop  <- data2haplohh(hap_file = filesPOST$filesPOST[i],
          vcf_reader = "data.table", verbose = F, polarize_vcf = F)
  
  scan_pop1 <- scan_hh(prepop)
  scan_pop2 <- scan_hh(postpop)
  
  chrXP_EHH <- ies2xpehh(
    scan_pop1,
    scan_pop2,
    popname1 = "pre",
    popname2 = "post",
    min_nhaplo = NA,
    standardize = TRUE,
    include_freq = FALSE,
    p.side = NA,
    p.adjust.method = "none",
    verbose = TRUE
  )
  
  write.csv(chrXP_EHH, file = paste("./REHH/chr_", filesPOST$chrom[i], "XP_EHH.csv", sep = ""))
  
  rm(prepop, postpop, scan_pop1, scan_pop2, chrXP_EHH)
}
```

Only the core chromosomes had enough markers for the above process.

3) Concatenate all output files into one
```{r}
files <- list.files(path ="./REHH/", pattern = "*XP_EHH.csv", full.names = T, recursive = T)

dflist <- lapply(files, function(i){
 df <- read.csv(i)
 return(df)
})
EHH_CoreChroms <- do.call(rbind, dflist)
EHH_CoreChroms <- EHH_CoreChroms[,-1]

# With NA's
write.csv(EHH_CoreChroms, file = "./REHH/core_withNA_EHH.csv")
# Removed NA's

EHH_CoreChroms_noNA <- na.omit(EHH_CoreChroms)
write.csv(EHH_CoreChroms_noNA, file = "./REHH/core_chroms_EHH.csv")
```

```{r}
unique(EHH_CoreChroms$CHR)

chromorder <- c("WAI332_chr_01","WAI332_chr_02","WAI332_chr_04","WAI332_chr_05","WAI332_chr_06","WAI332_chr_07","WAI332_chr_08","WAI332_chr_09","WAI332_chr_10", "WAI332_chr_11", "WAI332_chr_12", "WAI332_chr_13")
```
____________________________________

```{r}
EHHdata <- read.csv("./data_sheets/XPEHH_na.rm.csv", header = T)
```

```{r}
chr1 <- EHHdata[EHHdata$CHR=="WAI332_chr_01",]
chr5 <- EHHdata[EHHdata$CHR=="WAI332_chr_05",]
```

# both pval and raw
```{r}
ggplot()+
  geom_point(data = chr1, aes(x=POSITION/1000000, y=XPEHH_pre_post), colour="darkslategrey")+
  geom_line(data=chr1, aes(x=POSITION/1000000, y=LOGPVALUE), colour="red")
```

# just pval
```{r}
ggplot()+
  geom_line(data=chr5, aes(x=POSITION/1000000, y=LOGPVALUE), colour="darkslategrey")
```


# manahhtan style of pval WG
```{r}

 ggplot(EHHdata, aes(x=CHR, y=LOGPVALUE,  color=CHR))+
  geom_jitter()+
    theme(legend.position="none")+
    xlab("Chromosome")+
    ggtitle("XP-EHH P vals")
```

# manahhtan style of raw WG
```{r}

 ggplot(EHHdata, aes(x=CHR, y=XPEHH_pre_post,  color=CHR))+
  geom_jitter()+
    theme(legend.position="none")+
    xlab("Chromosome")+
    ggtitle("XP-EHH raw")
```
